.github/workflows/test-gpu-dev.yml

```diff
@@ -0,0 +1,66 @@
+name: GPU-CI-DEV
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types:
+      - labeled
+      - opened
+      - synchronize
+
+# Cancel the job if new commits are pushed
+# https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: flying-sheep/check@v1
+        with:
+          success: ${{ github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'run-gpu-ci') }}
+  test:
+    name: GPU Tests
+    needs: check
+    runs-on: "cirun-aws-gpu--${{ github.run_id }}"
+    timeout-minutes: 30
+
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Nvidia SMI sanity check
+        run: nvidia-smi
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install UV
+        uses: hynek/setup-cached-uv@v2
+        with:
+          cache-dependency-path: pyproject.toml
+
+      - name: Install rapids-singlecell
+        run: uv pip install --system -e .[test,rapids12] --extra-index-url=https://pypi.nvidia.com
+
+      - name: Install scanpy-dev
+        run: uv pip install --system git+https://github.com/scverse/scanpy.git
+
+      - name: Install anndata-dev
+        run: uv pip install --system git+https://github.com/scverse/anndata.git
+
+
+      - name: Pip list
+        run: pip list
```

Comment:
I don’t think this is really necessary, `uv` tells you what it installed, right?